src/util.ts

```diff
@@ -953,3 +989,27 @@ export function isTokenDetectionEnabledForNetwork(chainId: string): boolean {
     chainId,
   );
 }
+
+/**
+ * Utility method to log if error is a common fetch error and otherwise rethrow it.
+ *
+ * @param error - Caught error that we should either rethrow or log to console.
+ * @param codesToCatch - array of error codes for errors we want to catch and log in a particular context
+ */
+export function logOrRethrowError(error: any, codesToCatch?: number[]) {
```

Comment:
`any` implies that `error` can be `null` or `undefined`. Should we check `error` for that?

Reply:
I believe this can/should be typeof `Error`

Reply:
done here: https://github.com/MetaMask/controllers/pull/805/commits/d29d97cd7a86c5bd82e20ac2d73a0f34ff43171c